### PR TITLE
[DV] Update ibex log regex

### PR DESCRIPTION
**Pull request by udinator**
Log parsing update to match the new modifications to the tracer.
Will not merge until #360 is merged, to avoid breaking any current flows.

### CODE REVIEW

File: dv/uvm/sim.py [hunk 1]
```diff
@@ -178,8 +178,8 @@ def compare(test_list, iss, output_dir, verbose):
       logging.info("Comparing %s/DUT sim result : %s" % (iss, elf))
       run_cmd(("echo 'Test binary: %s' >> %s" % (elf, report)))
       uvm_log = ("%s/rtl_sim/%s.%d/sim.log" % (output_dir, test['test'], i))
-      rtl_log = ("%s/rtl_sim/%s.%d/trace_core_00000000.log" % (output_dir, test['test'], i))
-      rtl_csv = ("%s/rtl_sim/%s.%d/trace_core_00000000.csv" % (output_dir, test['test'], i))
+      rtl_log = ("%s/rtl_sim/%s.%d/trace_core__00000000.log" % (output_dir, test['test'], i))
+      rtl_csv = ("%s/rtl_sim/%s.%d/trace_core__00000000.csv" % (output_dir, test['test'], i))
```
Comment (imphil) [inline dv/uvm/sim.py]: Oh, that double `__` looks unintentional. I'll double-check.

Comment (imphil) [reply dv/uvm/sim.py]: Opened https://github.com/lowRISC/ibex/pull/367 to get this fixed.

File: dv/uvm/riscv_dv_extension/ibex_log_to_trace_csv.py [hunk 2]
```diff
@@ -31,9 +31,9 @@ def process_ibex_sim_log(ibex_log, csv):
             if re.search("ecall", line):
               break
             # Extract instruction infromation
```
Comment (imphil) [inline dv/uvm/riscv_dv_extension/ibex_log_to_trace_csv.py]: unrelated nit: infromation -> information

File: dv/uvm/riscv_dv_extension/ibex_log_to_trace_csv.py [hunk 3]
```diff
@@ -31,9 +31,9 @@ def process_ibex_sim_log(ibex_log, csv):
...
-            m = re.search(r"^\s*(?P<time>\d+)\s+(?P<cycle>\d+) " \
-                          "(?P<pc>[0-9a-f]+) (?P<bin>[0-9a-f]+) (?P<instr>.*)" \
-                          "x(?P<rd>\d+)=0x(?P<val>[0-9a-f]+)", line)
+            m = re.search(r"^\s*(?P<time>\d+)\s+(?P<cycle>\d+)\s+" \
+                          "(?P<pc>[0-9a-f]+)\s+(?P<bin>[0-9a-f]+)\s+(?P<instr>.*)" \
+                          ".*x(?P<rd>\d+)=0x(?P<val>[0-9a-f]+)", line)
```
Comment (imphil) [inline dv/uvm/riscv_dv_extension/ibex_log_to_trace_csv.py]: Now that the entries are tab-separated, you can use `line.strip().split("\t")`

That's code I've been using:


```python
def get_instructions_from_trace(trace_file):
    instructions = {}
    with open(trace_file) as f:
        line_cnt = -1
        for line in f:
            line_cnt += 1
            if line_cnt == 0:
                continue
            elements = line.strip().split("\t")
            time = elements[0]
            cycle = elements[1]
            pc = int(elements[2], base=16)
            instr = int(elements[3], base=16)
            opcode = elements[4]
            desc = ''
            if len(elements) > 5:
                desc = elements[5]
            instructions[pc] = {'instr': instr, 'opcode': opcode, 'desc': desc, 'pc': pc}
    return instructions
```

File: dv/uvm/sim.py (hunk 1 above)
Comment (udinator) [reply dv/uvm/sim.py]: Ah sure, I've approved your related filename PR and reverted the filename in this script.

File: dv/uvm/riscv_dv_extension/ibex_log_to_trace_csv.py (hunk 2 above)
Comment (udinator) [reply dv/uvm/riscv_dv_extension/ibex_log_to_trace_csv.py]: Good catch, thanks for pointing it out.

File: dv/uvm/riscv_dv_extension/ibex_log_to_trace_csv.py (hunk 3 above)
Comment (udinator) [reply dv/uvm/riscv_dv_extension/ibex_log_to_trace_csv.py]: I think keeping our regex is better for our post-simulation comparison flows, as we only really care about the RD register and the rd_wdata, I think the regex allows us to extract that information more easily for our use-case

Comment (imphil) [reply dv/uvm/riscv_dv_extension/ibex_log_to_trace_csv.py]: Sure, whatever works for your use case. (If you want more structured data, a separate tracer that makes use of the RVFI interface might be more future-proof, but that's not for now.)

**Pull request merged**